quickwit-cli/src/lib.rs

```diff
@@ -293,6 +301,37 @@ pub async fn delete_index_cli(args: DeleteIndexArgs) -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn garbage_collect_index_cli(args: GarbageCollectIndexArgs) -> anyhow::Result<()> {
+    debug!(
+        index_uri = %args.index_uri,
+        "garbage-collect-index"
+    );
+    quickwit_telemetry::send_telemetry_event(TelemetryEvent::GarbageCollect).await;
+
+    let (metastore_uri, index_id) =
+        extract_metastore_uri_and_index_id_from_index_uri(&args.index_uri)?;
+    let affected_files = garbage_collect_index(metastore_uri, index_id).await?;
```

Comment:
Should we rename `deleted_files` or is it more subtle than this?

Reply:
No, it's not especially since we don't have `dry-run` like `delete` command